lib/puppet/reports/prometheus.rb

```diff
@@ -89,6 +89,15 @@ def process
 # TYPE puppet_report gauge
 EOS
 
+    if defined?(transaction_completed) && ([true, false].include? transaction_completed)
+      completed = transaction_completed == true ? 1 : 0
+      new_metrics["puppet_transaction_completed{#{common_values.join(',')}}"] = completed
+      definitions << <<-EOS
+# HELP puppet_transaction_completed transaction completed status of the last puppet run
```

Comment:
Could you make the indentation here similar to the other metrics above?
Other than that, LGTM.  Could you also rebase into a single commit?

Reply:
Hi Alex,

I will rebase.
About the indent, as this is in an if block the indent is one level extra, with the HERE doc being left aligned. If that is not ok i'm not sure what you mean.

Stefan.

Reply:
It's fine, just none of the other HEREDOCs are left aligned.  But... maybe they all should be, since they're not using ruby 2.3 style heredocs (which aren't available in jruby used by puppetserver IIRC.)